### PR TITLE
Fix throwing non-new Errors and literals

### PR DESCRIPTION
**Pull request by ddbeck**
Blocked on https://github.com/web-platform-dx/web-features/pull/1312. From https://github.com/web-platform-dx/web-features/pull/1312#discussion_r1665624028

Note to self: after merge, rebase and run `npm run test:lint -- --fix`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,27 +18,31 @@
...
+    "test:format": "prettier --check .",
+    "test:lint": "npx eslint .",
     "test:schema": "tsx scripts/schema.ts",
     "test:specs": "tsx scripts/specs.ts",
-    "test:format": "prettier --check .",
-    "test:coverage": "npm run --workspaces test:coverage",
-    "format": "npx prettier --write .",
-    "feature-init": "tsx scripts/feature-init.ts",
+    "test": "npm run test:caniuse -- --quiet && npm run test:schema && npm run test:specs && npm run test:format && npm run test:dist && npm run test --workspaces && npm run test:lint",
     "update-drafts": "tsx scripts/update-drafts.ts"
```
Comment (ddbeck) [inline package.json:33]: I sorted these, sorry for the chaos diff.

File: package.json [hunk 2]
```diff
@@ -18,27 +18,31 @@
...
+    "feature-init": "tsx scripts/feature-init.ts",
+    "format": "npx prettier --write .",
     "schema-defs:write": "npm run schema-defs -- --out ./schemas/defs.schema.json",
-    "test": "npm run test:caniuse -- --quiet && npm run test:schema && npm run test:specs && npm run test:format && npm run test:dist && npm run test --workspaces",
+    "schema-defs": "ts-json-schema-generator --tsconfig ./tsconfig.json --type FeatureData --path ./types.ts --id defs",
     "test:caniuse": "tsx scripts/caniuse.ts",
+    "test:coverage": "npm run --workspaces test:coverage",
     "test:dist": "tsx scripts/dist.ts --check",
+    "test:format": "prettier --check .",
+    "test:lint": "npx eslint .",
```
Comment (ddbeck) [inline package.json:29]: New!

File: package.json [hunk 3]
```diff
@@ -18,27 +18,31 @@
...
     "test:schema": "tsx scripts/schema.ts",
     "test:specs": "tsx scripts/specs.ts",
-    "test:format": "prettier --check .",
-    "test:coverage": "npm run --workspaces test:coverage",
-    "format": "npx prettier --write .",
-    "feature-init": "tsx scripts/feature-init.ts",
+    "test": "npm run test:caniuse -- --quiet && npm run test:schema && npm run test:specs && npm run test:format && npm run test:dist && npm run test --workspaces && npm run test:lint",
```
Comment (ddbeck) [inline package.json:32]: Changed!

File: packages/compute-baseline/src/browser-compat-data/feature.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ export class Feature {
 
   constructor(id: string, featureData: unknown) {
     if (!isFeatureData(featureData)) {
-      throw `${id} is not valid feature`;
+      throw new Error(`${id} is not valid feature`);
```
Comment (ddbeck) [inline packages/compute-baseline/src/browser-compat-data/feature.ts:30]: Previously thrown as a string, not an error

**Review by ddbeck — COMMENTED** [review]
Self-review to highlight relevant bits.

**Review by foolip — APPROVED** [review]

**Pull request merged**